sys/vm/vm_map.c

```diff
@@ -5427,7 +5427,13 @@ vmspace_exec(struct proc *p, vm_offset_t minuser, vm_offset_t maxuser)
 	    ("Unrepresentable length for new vmspace"));
 
 	user_length = CHERI_REPRESENTABLE_LENGTH(user_length);
-	minuser_cap = (vm_pointer_t)cheri_capability_build_user_rwx(
+	/*
+	 * XXX: Use the unchecked version here because the map is empty
+	 * at this point.
+	 *
+	 * XXX: It seems like this should be a pmap call.
```

Comment:
Or maybe a sv_foo given that `maxuser` comes from there?

Reply:
Yeah, there's an argument for that. It might be that sv_minuser and sv_maxuser want to become capabilities. Things are made a bit weird by the fact that minuser is computed in exec_new_vmspace. I'm not really sold on `map_at_zero` being a global setting...